src/bank.rs

```diff
@@ -229,45 +235,39 @@ impl Bank {
 
     /// Forget all signatures. Useful for benchmarking.
     pub fn clear_signatures(&self) {
-        for sigs in &mut self.last_ids_q.write().unwrap().last_ids_sigs.values_mut() {
+        for sigs in &mut self.last_ids.write().unwrap().sigs.values_mut() {
             sigs.0.clear();
         }
     }
 
-    /// Return the position of the last_id in the last_id_queue starting from the back
-    /// If the last_id is not found last_id_queue.len() is returned
-    fn compute_entry_id_age(last_id_queue: &VecDeque<Hash>, entry_id: Hash) -> Option<usize> {
-        for (i, id) in last_id_queue.iter().rev().enumerate() {
-            if *id == entry_id {
-                return Some(i);
-            }
-        }
-        None
-    }
     /// Check if the age of the entry_id is within the max_age
     /// return false for any entries with an age equal to or above max_age
-    fn check_entry_id_age(last_id_queue: &VecDeque<Hash>, entry_id: Hash, max_age: usize) -> bool {
-        match Self::compute_entry_id_age(last_id_queue, entry_id) {
-            Some(age) if age < max_age => true,
+    fn check_entry_id_age(last_ids: &LastIds, entry_id: Hash, max_age: usize) -> bool {
+        let entry = last_ids.sigs.get(&entry_id);
+
+        match entry {
+            Some(entry) => ((last_ids.nth - entry.2) as usize) < max_age,
```

Comment:
If you replace the implementation of `compute_entry_id_age` with this expression, are any of its tests still relevant?

Reply:
I don't think so, which is why I removed it